students/jared_mulholland/lesson_3/mail_room.py

```diff
@@ -0,0 +1,101 @@
+
+    thankyou = f"""\nDear {name.split()[0]}, \n\nThank you for your generous donation of ${new_donation:,.2f}. Thanks to you we will finally be able to begin construction 
+on the {name.split()[1]} Memorial Children's wing at Dark Place Hospital\n\nYours Truly,\nJared Mulholland"""
+
+    print(thankyou)
+    return donors, donations
+
+
+"""
+CREATE A REPORT
+If the user (you) selected “Create a Report”, print a list of your donors, sorted by total historical donation amount.
+Include Donor Name, total donated, number of donations and average donation amount as values in each row. You do not need to print out all their donations, just the summary info.
+Using string formatting, format the output rows as nicely as possible. The end result should be tabular (values in each column should align with those above and below)
+After printing this report, return to the original prompt.
+At any point, the user should be able to quit their current task and return to the original prompt.
+From the original prompt, the user should be able to quit the script cleanly.
+"""
+import numpy as np
+
+def takeSecond(elem):
+    return elem[1]
+
+def create_report():
+    rows = []
+    for donor in donors:
+        rows.append([donor, sum(donations[donors.index(donor)]),len(donations[donors.index(donor)]),np.mean(donations[donors.index(donor)])])
+
+    rows.sort(key = takeSecond, reverse = True)
+
+    print('{:<20s} |{:>15s}|{:>12s} |{:>15s}'.format('Donor Name','Total Given','Num Gifts','Average Gift'))
+    for row in rows:
+        print('{:<20s} ${:15,.2f} {:12d} ${:15,.2f}'.format(*row))
+
+
+
+"""
+PROGRAM OVERVIEW
+Write a small command-line script called mailroom.py. This script should be executable. The script should accomplish the following goals:\
+-It should have a data structure that holds a list of your donors and a history of the amounts they have donated. 
+    This structure should be populated at first with at least five donors, with between 1 and 3 donations each.
+-You can store that data structure in the global namespace.
+-The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”)
+"""
+
+donors = ['Bill Gates','Mark Zuckerberg','Jeff Bezos','Paul Allen','Mookie Betts']
+donations = [[10000,12000],[20000,150000],[800],[100,500,200],[1000000,1500000,1750000]]
+
+response = input("What would you like to do (enter 'Send a Thank You', 'Create a Report' or 'Quit'): ")
+
+while response.lower().replace(" ","") != 'quit':
+    if response.lower().replace(" ","") == 'sendathankyou':
+        print('\n')
+        send_thankyou()
+        response = input("\nWhat would you like to do (enter 'Send a Thank You', 'Create a Report' or 'Quit'): ")
+    
+    elif response.lower().replace(" ","") == 'createareport':
+        print('\n')
+        create_report()
+        response = input("\nWhat would you like to do (enter 'Send a Thank You', 'Create a Report' or 'Quit'): ")
```

Comment:
couple comments about this entry point:
* instead of calling `replace` 3 times.. call it once at the end of `input` or into a separate variable
** instead of calling `replace` you can use `strip()`
* use numbers instead of strings - strings are error prone and not user friendly, your options can be 1, 2 and 3
* you don't need to call `input` function 3 times, just once

see example:
```
while True:
    response = input("1 - Send Thank You, ... ")
    if response == "1":
        send_thank_you()
    elif response == "2":
        create_report()
    elif response == "3":
        break
```